File: src/src_core/fmg/core/img/IAnimator.java
package fmg.core.img;

import java.util.function.Consumer;

/** Perform tasks for calculating / changing the subscribers animation. Executed in the current UI thread */
public interface IAnimator {

   void subscribe(Object subscriber, Consumer<Long /** time in ms from start subscribe */> subscriberCallbackMethod);
   void pause(Object subscriber);
   void unsubscribe(Object subscriber);

}

File: src/src_core/fmg/core/img/IAnimator.cs
using System;

namespace fmg.core.img {

    /// <summary> Perform tasks for calculating / changing the subscribers animation. Executed in the current UI thread </summary>
    public interface IAnimator {

        void Subscribe(object subscriber, Action<TimeSpan /** time in ms from start subscribe */> subscriberCallbackMethod);
        void Pause(object subscriber);
        void Unsubscribe(object subscriber);

    }

}
